src/badge/withBadge.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+import Badge from './Badge';
+
+const styles = StyleSheet.create({
+  badge: {
+    borderRadius: 9,
+    height: 18,
+    minWidth: 0,
+    width: 18
+  },
+  badgeContainer: {
+    position: 'absolute'
+  },
+  badgeText: {
+    fontSize: 10,
+    paddingHorizontal: 0
+  }
+});
+
+const withBadge = (value, options = {}) => WrappedComponent => {
+  class WithBadge extends React.Component {
+    render() {
+      const { left = 0, bottom = 0, top = -5, right = 0, hidden = !value, ...badgeProps } = options;
+      const badgeValue = typeof value === 'function' ? value(this.props) : value;
```

Comment:
When would this be a function?

Reply:
When you want to get the badge value from the props.

```jsx
withBadge(props => props.notifications.length)…
```

Still the same use case with the `connect` thing in the docs

Reply:
Gotcha! I don't use decorators, so i had no idea! Thanks though